interfaces/console/src/app/console/sites/[id]/page.tsx

```diff
@@ -7,47 +7,233 @@
+  spectrum: '',
+  siteName: '',
+  latitude: NaN,
+  longitude: NaN,
+  network: '',
+};
+const defaultSite: SiteDto = {
+  id: '',
+  accessId: '',
+  backhaulId: '',
+  createdAt: '',
+  installDate: '',
+  isDeactivated: false,
+  latitude: 0,
+  location: '',
+  longitude: 0,
+  name: '',
+  networkId: '',
+  powerId: '',
+  spectrumId: '',
+  switchId: '',
+};
 interface SiteDetailsProps {
   params: {
     id: string;
   };
 }
 const Page: React.FC<SiteDetailsProps> = ({ params }) => {
   const { id } = params;
-  const [open, setOpen] = useState(false);
-  const [restartDialogOpen, setRestartDialogOpen] = useState(false);
-
-  const handleClose = () => {
-    setOpen(false);
-  };
-  const handleRestartSite = (site: { id: string; name: string }) => {
-    console.log(`Restarting site: ${id}`);
-    setRestartDialogOpen(true);
+  const [site, setSite] = useState<TSiteForm>(SITE_INIT);
+  const [activeSite, setActiveSite] = useState<SiteDto>(defaultSite);
+  const [openSiteConfig, setOpenSiteConfig] = useState(false);
+  const [componentsList, setComponentsList] = useState<any[]>([]);
+  const [selectedSiteId, setSelectedSiteId] = useState<string | null>(null);
+  const [sitesList, setSitesList] = useState<SiteDto[]>([]);
+  const { setSnackbarMessage, network } = useAppContext();
+  const router = useRouter();
+  const handleSiteConfigOpen = () => {
+    setOpenSiteConfig(true);
   };
 
-  const handleAddSite = () => {
-    setOpen(true);
+  const handleCloseSiteConfig = () => {
+    setOpenSiteConfig(false);
   };
 
-  const handleSiteInstallation = (formData: any) => {
-    console.log('Function not implemented.', formData);
+  const [addSite, { loading: addSiteLoading }] = useAddSiteMutation({
+    onCompleted: (res) => {
+      setSnackbarMessage({
+        id: 'add-site-success',
+        message: 'Site added successfully!',
+        type: 'success' as AlertColor,
+        show: true,
+      });
+    },
+    onError: (error) => {
+      setSnackbarMessage({
+        id: 'add-subscriber-error',
+        message: error.message,
+        type: 'error' as AlertColor,
+        show: true,
+      });
+    },
+  });
+  const getCurrentDateInISOFormat = () => {
+    const date = new Date();
+    return date.toISOString().split('T')[0] + 'T00:00:00Z';
   };
-  const handleRestartDialogClose = () => {
-    setRestartDialogOpen(false);
+  const { loading: sitesLoading } = useGetSitesQuery({
+    skip: !network.id,
+    variables: {
+      networkId: network.id,
+    },
+    onCompleted: (res) => {
+      setSitesList(res.getSites.sites);
+    },
+    onError: (error) => {
+      setSnackbarMessage({
+        id: 'fetching-sites-msg',
+        message: error.message,
+        type: 'error' as AlertColor,
+        show: true,
+      });
+    },
+  });
+  const handleSiteConfiguration = async (data: any) => {
+    const variables = {
+      access_id: data.access,
+      backhaul_id: data.backhaul,
+      install_date: getCurrentDateInISOFormat(),
```

Comment:
We are using `date-fns` package which can help us here. On the sites page i have replace it with 
`formatISO(new Date())` & formatISO is imported from `date-fns` like `import { formatISO } from 'date-fns';`